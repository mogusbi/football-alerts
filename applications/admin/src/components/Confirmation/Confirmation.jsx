import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import PropTypes from 'prop-types';
import React, {memo} from 'react';

function Confirmation ({closeHandler, confirmHandler, message, open, title}) {
  return (
    <Dialog
      aria-describedby='confirmation-dialog-description'
      aria-labelledby='confirmation-dialog-title'
      onClose={closeHandler}
      open={open}
    >
      <DialogTitle id='confirmation-dialog-title'>
        {title}
      </DialogTitle>
      <DialogContent>
        <DialogContentText id='confirmation-dialog-description'>
          {message}
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button
          color='primary'
          onClick={closeHandler}
          id='confirmation-dialog-cancel'
        >
          Cancel
        </Button>
        <Button
          color='primary'
          onClick={confirmHandler}
          id='confirmation-dialog-yes'
        >
          Yes
        </Button>
      </DialogActions>
    </Dialog>
  );
}

Confirmation.propTypes = {
  closeHandler: PropTypes.func.isRequired,
  confirmHandler: PropTypes.func.isRequired,
  message: PropTypes.string.isRequired,
  open: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired
};

export default memo(Confirmation);
