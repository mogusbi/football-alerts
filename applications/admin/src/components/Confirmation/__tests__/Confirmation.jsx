import {mount, shallow} from 'enzyme';
import React from 'react';
import Confirmation from '../Confirmation';

describe('Confirmation', () => {
  let closeHandler;
  let confirmHandler;
  let message;
  let open;
  let title;
  let wrapper;

  beforeEach(() => {
    closeHandler = jest.fn();
    confirmHandler = jest.fn();
    title = 'Test confirmation';
    message = 'This is a message';
  });

  describe('when open', () => {
    beforeEach(() => {
      open = true;

      wrapper = shallow(
        <Confirmation
          closeHandler={closeHandler}
          confirmHandler={confirmHandler}
          message={message}
          open={open}
          title={title}
        />
      );
    });

    it('should render component', () => {
      expect(wrapper.dive()).toMatchSnapshot();
    });

    it('should call closeHandler when cancelled', () => {
      const closeButton = wrapper.find('#confirmation-dialog-cancel');

      closeButton.simulate('click');

      expect(closeHandler).toHaveBeenCalledTimes(1);
    });

    it('should call closeHandler when confirmed', () => {
      const cancelButton = wrapper.find('#confirmation-dialog-yes');

      cancelButton.simulate('click');

      expect(confirmHandler).toHaveBeenCalledTimes(1);
    });
  });

  describe('when closed', () => {
    beforeEach(() => {
      open = false;

      wrapper = mount(
        <Confirmation
          closeHandler={closeHandler}
          confirmHandler={confirmHandler}
          message={message}
          open={open}
          title={title}
        />
      );
    });

    it('should render component', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
