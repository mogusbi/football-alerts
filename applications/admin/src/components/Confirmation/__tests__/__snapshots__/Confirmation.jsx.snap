// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation when closed should render component 1`] = `
<Confirmation
  closeHandler={[MockFunction]}
  confirmHandler={[MockFunction]}
  message="This is a message"
  open={false}
  title="Test confirmation"
>
  <WithStyles(Dialog)
    aria-describedby="confirmation-dialog-description"
    aria-labelledby="confirmation-dialog-title"
    onClose={[MockFunction]}
    open={false}
  >
    <Dialog
      PaperComponent={[Function]}
      TransitionComponent={[Function]}
      aria-describedby="confirmation-dialog-description"
      aria-labelledby="confirmation-dialog-title"
      classes={
        Object {
          "container": "MuiDialog-container-4",
          "paper": "MuiDialog-paper-5",
          "paperFullScreen": "MuiDialog-paperFullScreen-14",
          "paperFullWidth": "MuiDialog-paperFullWidth-13",
          "paperScrollBody": "MuiDialog-paperScrollBody-7",
          "paperScrollPaper": "MuiDialog-paperScrollPaper-6",
          "paperWidthLg": "MuiDialog-paperWidthLg-11",
          "paperWidthMd": "MuiDialog-paperWidthMd-10",
          "paperWidthSm": "MuiDialog-paperWidthSm-9",
          "paperWidthXl": "MuiDialog-paperWidthXl-12",
          "paperWidthXs": "MuiDialog-paperWidthXs-8",
          "root": "MuiDialog-root-1",
          "scrollBody": "MuiDialog-scrollBody-3",
          "scrollPaper": "MuiDialog-scrollPaper-2",
        }
      }
      disableBackdropClick={false}
      disableEscapeKeyDown={false}
      fullScreen={false}
      fullWidth={false}
      maxWidth="sm"
      onClose={[MockFunction]}
      open={false}
      scroll="paper"
      transitionDuration={
        Object {
          "enter": 225,
          "exit": 195,
        }
      }
    >
      <WithStyles(Modal)
        BackdropProps={
          Object {
            "transitionDuration": Object {
              "enter": 225,
              "exit": 195,
            },
          }
        }
        aria-describedby="confirmation-dialog-description"
        aria-labelledby="confirmation-dialog-title"
        className="MuiDialog-root-1"
        closeAfterTransition={true}
        disableBackdropClick={false}
        disableEscapeKeyDown={false}
        onClose={[MockFunction]}
        open={false}
        role="dialog"
      >
        <Modal
          BackdropComponent={[Function]}
          BackdropProps={
            Object {
              "transitionDuration": Object {
                "enter": 225,
                "exit": 195,
              },
            }
          }
          aria-describedby="confirmation-dialog-description"
          aria-labelledby="confirmation-dialog-title"
          className="MuiDialog-root-1"
          classes={
            Object {
              "hidden": "MuiModal-hidden-16",
              "root": "MuiModal-root-15",
            }
          }
          closeAfterTransition={true}
          disableAutoFocus={false}
          disableBackdropClick={false}
          disableEnforceFocus={false}
          disableEscapeKeyDown={false}
          disablePortal={false}
          disableRestoreFocus={false}
          hideBackdrop={false}
          keepMounted={false}
          manager={
            ModalManager {
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "modals": Array [],
            }
          }
          onClose={[MockFunction]}
          open={false}
          role="dialog"
        />
      </WithStyles(Modal)>
    </Dialog>
  </WithStyles(Dialog)>
</Confirmation>
`;

exports[`Confirmation when open should render component 1`] = `
<Dialog
  PaperComponent={[Function]}
  TransitionComponent={[Function]}
  aria-describedby="confirmation-dialog-description"
  aria-labelledby="confirmation-dialog-title"
  classes={
    Object {
      "container": "MuiDialog-container-4",
      "paper": "MuiDialog-paper-5",
      "paperFullScreen": "MuiDialog-paperFullScreen-14",
      "paperFullWidth": "MuiDialog-paperFullWidth-13",
      "paperScrollBody": "MuiDialog-paperScrollBody-7",
      "paperScrollPaper": "MuiDialog-paperScrollPaper-6",
      "paperWidthLg": "MuiDialog-paperWidthLg-11",
      "paperWidthMd": "MuiDialog-paperWidthMd-10",
      "paperWidthSm": "MuiDialog-paperWidthSm-9",
      "paperWidthXl": "MuiDialog-paperWidthXl-12",
      "paperWidthXs": "MuiDialog-paperWidthXs-8",
      "root": "MuiDialog-root-1",
      "scrollBody": "MuiDialog-scrollBody-3",
      "scrollPaper": "MuiDialog-scrollPaper-2",
    }
  }
  disableBackdropClick={false}
  disableEscapeKeyDown={false}
  fullScreen={false}
  fullWidth={false}
  maxWidth="sm"
  onClose={[MockFunction]}
  open={true}
  scroll="paper"
  transitionDuration={
    Object {
      "enter": 225,
      "exit": 195,
    }
  }
>
  <WithStyles(DialogTitle)
    id="confirmation-dialog-title"
  >
    Test confirmation
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <WithStyles(DialogContentText)
      id="confirmation-dialog-description"
    >
      This is a message
    </WithStyles(DialogContentText)>
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      id="confirmation-dialog-cancel"
      onClick={[MockFunction]}
    >
      Cancel
    </WithStyles(Button)>
    <WithStyles(Button)
      color="primary"
      id="confirmation-dialog-yes"
      onClick={[MockFunction]}
    >
      Yes
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</Dialog>
`;
